kotlin-native/runtime/src/launcher/cpp/launcher.cpp

```diff
@@ -30,8 +30,12 @@ using namespace kotlin;
 //--- Setup args --------------------------------------------------------------//
 
 OBJ_GETTER(setupArgs, int argc, const char** argv) {
+  if (argc > 0 && argv[0][0] != '\0') {
+    kotlin::programName = argv[0];
```

Comment:
How about doing `kotlin::programName = strndup(argv[0], 4096)`? No need to `free` the result, and we don't have to worry about lifetimes. `4096` is just some random not-too-small not-too-big number to protect us from `argv[0]` being too large. Technically, the OS has some limits already, but might as well protect ourselves.

Reply:
Ok, I applied your change.